scheds/rust/scx_rusty/src/bpf/cpumask.h

```diff
@@ -0,0 +1,199 @@
+extern const volatile u32 nr_cpu_ids;
+size_t mask_size;
+
+/* XXX HACK hardcoded MAX_CPUS / 8 from src/bpf/intf.h, right now this header is a mess */
+struct scx_cpumask {
+	union sdt_id tid;
+	u8 mask[64];
+};
+
+typedef struct scx_cpumask __arena * __arg_arena scx_cpumask_t;
+
+struct sdt_allocator scx_mask_allocator;
+
+__weak
+int scx_mask_init(__u64 total_mask_size)
+{
+	mask_size = total_mask_size;
+
+	return sdt_alloc_init(&scx_mask_allocator, mask_size + sizeof(union sdt_id));
+}
+
+static
+scx_cpumask_t scx_mask_alloc(void)
+{
+	struct sdt_data __arena *data = NULL;
+	scx_cpumask_t mask;
+	int i;
+
+	data = sdt_alloc(&scx_mask_allocator);
+	cast_kern(data);
+
+	mask = (scx_cpumask_t)data->payload;
+	mask->tid = data->tid;
+	bpf_for(i, 0, mask_size) {
+		mask->mask[i] = 0ULL;
+	}
+
+	return mask;
+}
+
+/*
+ * XXXETSAL: Ideally these functions would have a void return type,
+ * but as of 6.13 the verifier requires global functions to return a scalar.
+ */
+
+__weak
+int scx_mask_free(scx_cpumask_t __arg_arena mask)
+{
+	sdt_subprog_init_arena();
+
+	sdt_free_idx(&scx_mask_allocator, mask->tid.idx);
+	return 0;
+}
+
+
+__weak
+int scxmask_set_cpu(u32 cpu, scx_cpumask_t __arg_arena mask)
```

Comment:
There are three prefixes in use - `scx_cpumask`, `scx_mask` and `scxmask`. If this can be generalized in terms of number of bits (which we do need), maybe use `scx_bitmap`? In kernel, cpumasks are just wrapping bitmaps too.